content/articles/implementing-flask-api-using-linked-list/index.md

```diff
@@ -0,0 +1,520 @@
+
+For more in-depth discussions on various types of linked lists visit this [article](https://www.javatpoint.com/ds-types-of-linked-list#:~:text=The%20following%20are%20the%20types,Circular%20Linked%20list).
+
+Let's dive into implementing the linked list.
+
+Open the terminal and create a directory called `FlaskAPI`. Change into the directory and create a linked list file as shown below:
+
+```bash
+mkdir FlaskAPI
+cd FlaskAPI
+touch linked_list.py
+```
+In the linked list file write the code below:
+
+```python
+class Node:
+def __init__(self, data=None, next_node=None):
+self.data = data
+self.next_node = next_node
+
+class LinkedList:
+def __init__(self):
+self.head = None
+self.last_node = None
+
+def to_list(self):
+  pass
+
+def insert_beginning(self, data):
+pass
+
+def insert_at_end(self, data):
+pass
+
+def print_ll(self):
+pass
+```
+
+We first begin by creating a Node class which is initialized by three parameters `self`, `data`, and `next_node`. The `self` keyword in Python represents a specific instance of a class and is used to access the attributes and methods of the class in Python.
+
+The `self.data` and `self.next_node` either default to None or equals the value pass to data or next_node during the time of instantiation. The `__init__()` method is a constructor and it allows the class to initialize its attributes.
+
+The `LinkedList()` class constructor takes `head` and `last_node` as its attribute and assign them to None. We then create four methods `to_list()`, `insert_beginning`, `insert_at_end` ,and `print_ll` which we'll implement in a moment.
+
+With that said, let's visualize how a linked list works:
+
+![linked_list](/engineering-education/implementing-flask-api-using-linked-list/linked-list.jpg)
+
+The rectangles in the image above represent nodes in a linked list and each node has two separate compartments.
+
+The left compartment represents the data and has the string value Data in it. The right compartment represents the pointer that points to the next node.
+
+If we look at our Node class we see it has the data and next_node pointer as well. The `LinkedList()` wrapper class only helps us keep track of the head of our linked list.
+
+From the image above the first node in the linked list represents the head. So for instance, if we want to add to the head of our linked list as we'll see in a bit it will be easier to do so.
+
+The `toList()` takes a linked list object and converts it into a list. The implementation is shown below:
+
+```python
+def to_list(self):
+  l = []
+  if self.head is None:
+    return l
+  node = self.head
+  while node:
+    l.append(node.data)
+    node = node.next_node
+  return l
+```
+
+First, we create an empty list. Then proceed to check if the head of the linked list is `None`.
+
+If so, we return the empty list otherwise it's appended onto the list and the node is assigned to the node's next node. The list is then returned.
```

Comment:
```suggestion
If so, we return the empty list. Otherwise it is appended onto the list and the node is assigned to the node's next node. The list is then returned.
```